jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -52,13 +52,13 @@ private Path checkAliasPath()
 
         /* Catch situation where the Path class has already normalized
          * the URI eg. input path "aa./foo.txt"
-         * from an #addPath(String) is normalized away during
+         * from an #resolve(String) is normalized away during
          * the creation of a Path object reference.
-         * If the URI is different then the Path.toUri() then
+         * If the URI is different from the Path.toUri() then
          * we will just use the original URI to construct the
          * alias reference Path.
          */
-        if (!URIUtil.equalsIgnoreEncodings(uri, path.toUri()))
+        if (!URIUtil.equalsIgnoreEncodings(uri, normalize(path)))
```

Comment:
why do we need the expense of normalization here? `equalsIgnoreEncoding` already handles the /// differences.
What case is normalization catching?

Reply:
The `normalize(Path)` corrects ...

* `file:/path/to/foo.dat` to `file:///path/to/foo.dat`
* `jar:file:/path/to/foo.jar!/bar.dat` to `jar:file:/path/to/foo.jar!/bar.dat`
* `file:///path/to/dir` to `file:///path/to/dir/`

So that the `equalsIgnoreEncoding` is happy.
While the first 2 corrections are now handled by `equalsIgnoreEncoding` the last one isn't

Reply:
Is 2nd correction correct?

Reply:
Good catch, typo

Reply:
The prior response is now a bit out of date, the `normalize(Path)` method doesn't do the `file:/` behavior, only the trailing slash behavior now in this PR.

Reply:
But back to my question... Why normalize a temporary URI when equalsIgnoreEncoding doesn't notice /// or /?  It compares the authority and not the URI string.

Reply:
The old `normalize(Path)` did the `file:/...` to `file:///`, as the `URI.equals(URI)` check is capable of evaluating that successfully.

The new `normalize(Path)` doesn't modify the `file:///` portion anymore.

Both the old and new `normalize(Path)` still enforces the trailing slash on the URI if the path is a directory.
This is done to match up with the forced trailing slash done in the `PathResource(URI)` constructor.

Reply:
This trailing slash behavior depends on the Path existing and being a directory.

One quirk of this behavior (and not the only one) ...
If the Resource.uri has `.../Unknown/` as a result of `Resource.resolve("Unknown/")` and the directory doesn't exist, then this results in a case where the input URI has a trailing slash, and the `path.toUri()` doesn't, even with the `normalize(Path)`.

Reply:
I still don't understand why we are normalizing here.   That evades aliases rather than detects them.

Reply:
I think you are getting caught up on the word "normalize" here.

Let me rename it, as that's NOT what it's doing.

Reply:
Renamed. with more comments about it's use in commit dce2da81daaa6b7a535093f8865f267deb1d9fc0